ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/utils/SimpleSeekableStream.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package eu.elixir.ega.ebi.dataedge.utils;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.http.HttpHeaders;
+
+import java.io.IOException;
+import java.net.URL;
+
+public class SimpleSeekableStream extends SeekableStream {
+
+    private static final int DEFAULT_CHUNK_SIZE = 4096;
+
+    private final URL uri;
+    private final int chunkSize;
+    private final long length;
+    private final OkHttpClient client;
+
+    private long position;
+    private byte[] buffer;
+    private long bufferPosition;
+
+    public SimpleSeekableStream(URL url, OkHttpClient client) throws IOException {
+        this(url, client, DEFAULT_CHUNK_SIZE);
+    }
+
+    public SimpleSeekableStream(URL url, OkHttpClient client, int chunkSize) throws IOException {
+        this(url, client, chunkSize, getContentLength(url, client));
+    }
+
+    protected static Long getContentLength(URL url, OkHttpClient client) throws IOException {
+        String contentLengthValue = client.newCall(new Request.Builder().url(url).head().build())
+                .execute()
```

Comment:
This is ok but optimistic, we need to control that the http return code is correct, otherwise we need eiter retry or fail fast with a 500 depending on the situation, and log any bad situation.